updater/lib/dependabot/updater/operations/create_security_update_pull_request.rb

```diff
@@ -59,17 +59,22 @@ def initialize(service:, job:, dependency_snapshot:, error_handler:)
         def perform
           Dependabot.logger.info("Starting security update job for #{job.source.repo}")
 
+          target_dependencies = dependency_snapshot.job_dependencies
+
+          # Raise an error if the package manager version is unsupported
+          dependency_snapshot.package_manager&.raise_if_unsupported!
+
           # Retrieve the list of initial notices from dependency snapshot
           @notices = dependency_snapshot.notices
           # More notices can be added during the update process
 
-          target_dependencies = dependency_snapshot.job_dependencies
-
           if target_dependencies.empty?
             record_security_update_dependency_not_found
           else
             target_dependencies.each { |dep| check_and_create_pr_with_error_handling(dep) }
           end
+        rescue StandardError => e
+          error_handler.handle_dependency_error(error: e, dependency: target_dependencies&.first)
         end
```

Comment:
When we have `unsupported error` for the package manager, we don't want to go over each dependency, instead we want to raise unsupported error for the whole process and handle error properly.